tmc-langs-ant/src/main/java/fi/helsinki/cs/tmc/langs/ant/AntPlugin.java

```diff
@@ -21,23 +32,164 @@ public String getLanguageName() {
+        Project buildProject = new Project();
+        buildProject.setUserProperty("ant.file", buildFile.getAbsolutePath());
+        buildProject.setProperty("javac.fork", "true");
+        buildProject.init();
+        buildProject.setBaseDir(path.toAbsolutePath().toFile());
+        File buildLog;
+        
+        try {
+            
+            DefaultLogger logger = new DefaultLogger();
+            buildLog = new File(path.toString(), "build_log.txt");
+            PrintStream errorPrintStream = new PrintStream(buildLog);
+            logger.setErrorPrintStream(errorPrintStream);
+            logger.setOutputPrintStream(System.out);
+            logger.setMessageOutputLevel(Project.MSG_ERR);
+            buildProject.addBuildListener(logger);
+            
+            try {
+                
+                buildProject.fireBuildStarted();
+                ProjectHelper helper = ProjectHelper.getProjectHelper();
+                buildProject.addReference("ant.projectHelper", helper);
+                helper.parse(buildProject, buildFile);
+                buildProject.executeTarget("compile-test");
+                buildProject.fireBuildFinished(null);
+                return true;
+                
+            } catch (BuildException e) {
+                
+                buildProject.fireBuildFinished(e);
+                buildRunResult = new RunResult(Status.COMPILE_FAILED, ImmutableList.copyOf(new ArrayList<TestResult>()),
+                        new ImmutableMap.Builder<String, byte[]>().put(SpecialLogs.COMPILER_OUTPUT,
+                                java.nio.file.Files.readAllBytes(buildLog.toPath())).build());
+                return false;
+                
+            }
+        } catch (IOException e) {
+            throw Throwables.propagate(e);
+        }
+    }
+
+    private List<String> buildTestScannerArgs(Path path, ClassPath classPath, String... args) {
+        List<String> scannerArgs = new ArrayList<>();
+
+        scannerArgs.add("java");
+        scannerArgs.add("-cp");
+
+        if (classPath == null) {
+            scannerArgs.add(generateClassPath(path).toString());
+        } else {
+            scannerArgs.add(classPath.toString());
+        }
+
+        scannerArgs.add("fi.helsinki.cs.tmc.testscanner.TestScanner");
+        path = path.toAbsolutePath();
+        scannerArgs.add(path.toString() + testDir);
+
+        if (args != null) {
+            for (String arg : args) {
+                scannerArgs.add(arg);
+            }
+        }
+
+        return scannerArgs;
+    }
+
+    private List<String> buildTestRunnerArgs(Path path) {
+        List<String> runnerArgs = new ArrayList<>();
+        List<String> testMethods;
+
+        runnerArgs.add("java");
+        runnerArgs.add("-Dtmc.test_class_dir=" + path.toString() + testDir);
+        runnerArgs.add("-Dtmc.results_file=" + path.toString() + resultsFile);
+        //runnerArgs.add("-Dfi.helsinki.cs.tmc.edutestutils.defaultLocale=" + locale);
```

Comment:
I wasn't sure if we need to set the locale, so I left it there for now. But as with testScannerArgs, this method as well should be deleted in the future.